### ISSUE TITLE
Bug: [prefer-readonly-parameter-types] should not error when rule is used

### ISSUE THREAD

**Issue by SerkanSipahi** [typescript-eslint/typescript-eslint#10502]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.tsx&code=FBA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge3oFsbDaOAIwBWiMnSKl0YAAwAacNjyTyVGvWSMWrQQEMmtfMlpkO3bohbT5inARKrqdBszad6AD2KMylOvsJrBWwoe1JkCicNLTcuRF4iWldEaAAzXTJEINtQlQi1Z01Xdi5YJlhkRAATT2JoFGRKDiZUDBsQ5Qd8qJdtOsRUlNp63Srm%2BH5iXWhdC3whvOkARkUAXxBVoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
There is not repo code. The rule itself produces the error.
```

### ESLint Config

```javascript
module.exports = [
  {
    files: ['**/*.ts', '**/*.tsx', '**/*.js', '**/*.jsx'],
    rules: {
      '@typescript-eslint/prefer-readonly-parameter-types': 1, // ❌ This leads to the error
      '@typescript-eslint/no-empty-object-type': 0,
      '@typescript-eslint/ban-ts-comment': 0,
      '@typescript-eslint/no-explicit-any': 0,
      '@typescript-eslint/no-empty-interface': 0,
      '@typescript-eslint/no-unused-expressions': 0,
      '@typescript-eslint/no-unused-vars': [
        'warn',
        {
          argsIgnorePattern: '^_',
          varsIgnorePattern: '^_',
          caughtErrorsIgnorePattern: '^_',
        },
      ],
      '@typescript-eslint/explicit-member-accessibility': [
        'error',
        {
          accessibility: 'no-public',
          overrides: {
            parameterProperties: 'explicit',
          },
        },
      ],
      'unused-imports/no-unused-imports': 'error',
    },
  },
];
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

 It should not throw an error when `@typescript-eslint/prefer-readonly-parameter-types` is set.

### Actual Result

Parsing error: "parserOptions.project" has been provided for @typescript-eslint/parser.
The file was not found in any of the provided project(s): packages/foo/eslint.config.js

### Additional Info

The error is only thrown if `@typescript-eslint/prefer-readonly-parameter-types` is used and the value is set to `1’` I got the rules from here https://typescript-eslint.io/rules. Most rules don't seem to cause problems and some like ‘@typescript-eslint/prefer-readonly-parameter-types’ lead to an error. Is this a bug or is there something else to consider?

**Comment by kirkwaiblinger** [comment]
Hi @SerkanSipahi ! Pretty sure this is just a configuration issue with [setting up typed linting](https://typescript-eslint.io/getting-started/typed-linting). The error is essentially complaining that you've enabled `prefer-readonly-parameter-types`, a rule that requires type information, but type information is not configured for the file it's being run on (in this case, a js eslint config file). Please have a look through the linked guide and see if that resolves your issue.

**Comment by SerkanSipahi** [comment]
@kirkwaiblinger 

You were absolutely right—the parsing error was caused by the ['**/*.js', '**/*.jsx'] configuration in files. Lint configurations aren’t something I touch every day. Sometimes we're blind to the things we do from time to time even if we read the docs.

Thank you so much for your quick and helpful support!

**Comment by kirkwaiblinger** [comment]
Glad that set you on the right track 🙂
